### PR TITLE
Cleanup: gps and divesite related

### PR DESCRIPTION
**Pull request by janmulder**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
The main reason for this series is a huge and almost literal code duplication. See also commits 6f42ab4 and ee9531f.  The code to apply GPS locations from the location service was already in core, and used from mobile, but there was an almost literal copy in the desktop code. As it was a simple C block of code, removal was simple.

In addition, some simple, almost trivial things are added in the same area of the code. 

Signed-off-by: Jan Mulder <jlmulder@xs4all.nl>

### Changes made:
See individual commits.

### Related issues:
commits 6f42ab4 and ee9531f.

### Additional information:
No functional changes

### CODE REVIEW

File: desktop-widgets/subsurfacewebservices.cpp [hunk 1]
```diff
@@ -387,8 +387,7 @@ SubsurfaceWebServices::SubsurfaceWebServices(QWidget *parent, Qt::WindowFlags f)
 
 	if (userid.isEmpty() &&
 	    !same_string(prefs.cloud_storage_email, "") &&
-	    !same_string(prefs.cloud_storage_password, "") &&
-	    GpsLocation::hasInstance())
+	    !same_string(prefs.cloud_storage_password, ""))
 		userid = GpsLocation::instance()->getUserid(prefs.cloud_storage_email, prefs.cloud_storage_password);
```
Comment (dirkhh) [inline desktop-widgets/subsurfacewebservices.cpp]: Now that you no longer check if there's an instance, the code will crash if there is no instance.
I wouldn't call that code "useless"...

File: core/divesitehelpers.cpp [hunk 2]
```diff
@@ -34,10 +34,10 @@ void ReverseGeoLookupThread::run()
 	QNetworkRequest request;
 	QNetworkAccessManager *rgl = new QNetworkAccessManager();
 	QEventLoop loop;
-	QString mapquestURL("http://open.mapquestapi.com/nominatim/v1/reverse.php?format=json&accept-language=%1&lat=%2&lon=%3");
+	//QString mapquestURL("http://open.mapquestapi.com/nominatim/v1/reverse.php?format=json&accept-language=%1&lat=%2&lon=%3");
```
Comment (dirkhh) [inline core/divesitehelpers.cpp]: don't comment them out. just remove them. we had planned multiple services to lookup locations but that never went anywhere.

**Review by dirkhh — CHANGES_REQUESTED** [review]
You are introducing a potential crash. Not good.

File: desktop-widgets/subsurfacewebservices.cpp (hunk 1 above)
Comment (janmulder) [reply desktop-widgets/subsurfacewebservices.cpp]: commit removed.

File: core/divesitehelpers.cpp (hunk 2 above)
Comment (janmulder) [reply core/divesitehelpers.cpp]: ok

**Comment by janmulder** [comment]
review comment processed => merging

**Pull request merged**